app/controllers/core_data_connector/sso_controller.rb

```diff
@@ -0,0 +1,88 @@
+require 'typhoeus'
+require 'jwt'
+require 'jwt_auth/configuration'
+require 'jwt_auth/json_web_token'
+require 'uri'
+
+module CoreDataConnector
+  class SsoController < ApplicationController
+    DEFAULT_TOKEN_EXPIRATION = 24
+
+    # No user authentication
+    skip_before_action :authenticate_request
+
+    def login
+      base_url = ENV['SSO_TOKEN_URL']
+
+      form_data = {
+        code: request.params['code'],
+        client_id: ENV['SSO_CLIENT_ID'],
+        client_secret: ENV['SSO_CLIENT_SECRET'],
+        grant_type: 'authorization_code',
+        redirect_uri: request.original_url.split('?')[0]
+      }
+
+      token_req = Typhoeus::Request.new(
+        base_url,
+        headers: {'Content-Type'=> "application/x-www-form-urlencoded"},
+        method: :post,
+        body: form_data
+      )
+
+      token_res = token_req.run
+      parsed = JSON.parse(token_res.body)
+
+      decoded = JWT.decode(parsed['access_token'], nil, false)[0]
+
+      sso_metadata = HashWithIndifferentAccess.new decoded
+
+      sso_user = find_sso_user(sso_metadata)
+
+      unless sso_user
+        render json: { error: 'User not found.' }, status: :bad_request
```

Comment:
I think `:bad_request` translates to an HTTP 400 code. Would 401 be more appropriate?